### ISSUE TITLE
I can't install npm modules in Hedera NFT Minter App project

### ISSUE THREAD

**Issue by draconian0428** [hashgraph/hedera-nft-minter#43]
### Description

Hello everyone.
How are you?

I can't install npm i in this repo. I used several ways to install modules but same result.
Who could let me know what node version I should use for this repo?

Thank you.

### Steps to reproduce

1. Run the program

### Hedera network

testnet

### Version

18.16.0

### Operating system

Windows

**Comment by pathornteng** [comment]
I also get the error when trying to install dependencies

```hedera-nft-minter git:(main) npm install
npm ERR! code ERESOLVE
npm ERR! ERESOLVE could not resolve
npm ERR! 
npm ERR! While resolving: sass-loader@12.1.0
npm ERR! Found: node-sass@7.0.1
npm ERR! node_modules/node-sass
npm ERR!   dev node-sass@"^7.0.1" from the root project
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peerOptional node-sass@"^4.0.0 || ^5.0.0 || ^6.0.0" from sass-loader@12.1.0
npm ERR! node_modules/sass-loader
npm ERR!   dev sass-loader@"12.1.0" from the root project
npm ERR! 
npm ERR! Conflicting peer dependency: node-sass@6.0.1
npm ERR! node_modules/node-sass
npm ERR!   peerOptional node-sass@"^4.0.0 || ^5.0.0 || ^6.0.0" from sass-loader@12.1.0
npm ERR!   node_modules/sass-loader
npm ERR!     dev sass-loader@"12.1.0" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR! 
npm ERR! 
npm ERR! For a full report see:
npm ERR! /Users/pathorn/.npm/_logs/2023-07-29T07_20_47_942Z-eresolve-report.txt

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/pathorn/.npm/_logs/2023-07-29T07_20_47_942Z-debug-0.log```

**Comment by Neurone** [comment]
Same issue for me, I tested multiple configurations (from 12 to 20) and I found you need exactly `16.14.2` to make install work. Even `16.15` is not good and you get that error.

I suggest specifying the NodeJS version in the readme.

The problem derives from the root project defining a direct dependency from `node-sass@"^7.0.1"` but sass-loader supports up to `node-sass@"^6.0.0"` only.

To make it work, you can also increase the sass-loader version to `^13.3.2`, but I didn't tested the entire application so probably sticking with NodeJS 16.14.2 is the better option.

Once installed all the dependencies, though, I got many failed tests:

```sh
❯ npm run test

> hedera-nft-minter@1.0.0 test
> node --experimental-vm-modules node_modules/jest/bin/jest.js --config=jest.config.js

 FAIL  tests/utils/helpers/filterFormValuesToNFTMetadata.test.ts
  ● Filter form values to metadata › correct metadata

    expect(received).toEqual(expected) // deep equality

    - Expected  - 5
    + Received  + 2

    @@ -8,13 +8,10 @@
        "creator": "Tester",
        "description": "lorem ipsum",
        "format": "HIP412@1.0.0",
        "image": "ipfs://bafkreiahrxk5xvmuqn2jmpaj2oemm777fazloc43ltxnsivzwigxy4cyjm",
        "name": "Test",
    -   "properties": Array [
    -     Object {
    -       "label": "foo",
    -       "value": "bar",
    +   "properties": Object {
    +     "foo": "bar",
        },
    -   ],
        "type": "image/jpeg",
      }

      38 |
      39 |     expect(filterFormValuesToNFTMetadata({...values, files: ''}))
    > 40 |       .toEqual({
         |        ^
      41 |         name: 'Test',
      42 |         description: 'lorem ipsum',
      43 |         type: 'image/jpeg',

      at Object.<anonymous> (tests/utils/helpers/filterFormValuesToNFTMetadata.test.ts:40:8)

 PASS  tests/services/IPFS.test.ts
...
...
...
Test Suites: 12 failed, 29 passed, 41 of 58 total
Tests:       3 failed, 54 passed, 57 total
Snapshots:   0 total
Time:        362 s ^C
```

**Comment by norbert-kulus-arianelabs** [comment]
Hi @Neurone, and thanks for your assistance in addressing this issue.

In Pull Request  #49, I've significantly updated the project's dependencies. It is crucial to use the latest stable Node.js version, currently ``Node.js 18.18.0``, for optimal performance.

Please review and let me know if there are any further actions required.

**Comment by Neurone** [comment]
Hi @norbert-kulus-arianelabs , thanks for the updates. The installation with 18.18 now runs fine, but I still have several errors running the tests. I also tried with an Ubuntu 20.04, just in case ;) but the result is the same.

I think I can fix some of those tests (i.e., adding a `return` [here](https://github.com/hashgraph/hedera-nft-minter/blob/f3d3525a094c930e243ed54615a1fce4e1f0def3/src/components/shared/form/FieldWrapper.tsx#L120) or using `import { Long } from '@hashgraph/sdk/lib/long';` and then `autoRenewPeriod?: number | Long;` instead of `autoRenewPeriod?: number | import('long').Long;` in the [HTS.ts](https://github.com/hashgraph/hedera-nft-minter/blob/f3d3525a094c930e243ed54615a1fce4e1f0def3/src/services/HTS.ts#L69)) but I'm not sure about the other more complex tests.

Can you give them a look?

**Comment by norbert-kulus-arianelabs** [comment]
Thank you for your patience and for testing the installation with 18.18. I appreciate your effort in trying it on Ubuntu as well.

I've dived deeper into all the tests and identified additional mistakes that needed attention. I've already addressed these issues in #50. Your feedback has been instrumental in uncovering these problems, and I'm grateful for your contribution to improving the project's quality. If you encounter any more issues or have further suggestions, please don't hesitate to let me know.
